packages/happy-dom/src/event/events/CustomEvent.ts

```diff
@@ -17,7 +17,7 @@ export default class CustomEvent extends Event {
 		super(type, eventInit);
 
 		if (eventInit) {
-			this.detail = eventInit.detail || null;
+			this.detail = eventInit.detail != null ? eventInit.detail : null;
```

Comment:
```suggestion
			this.detail = eventInit.detail != null && eventInit.detail != undefined ? eventInit.detail : null;
```

Reply:
`detail` should not be able to be "undefined". In this scenario it should become `null`.

Reply:
The equality check covers null and undefined cases. There is no further check needed.

Is there any documentation stating we have to use null in CustomEvent.detail? I am just wondering because in our code base we are also using undefined. Would be good to know if it is wrong.

Reply:
According to [spec](https://dom.spec.whatwg.org/#interface-customevent), the default value is "null". Passing "undefined" is considered as not passing any value, and it should then use the default value.

I have also looked how the browser behaves and it will set detail to "null" if we pass "undefined" as value. 

```javascript
var customEvent = new CustomEvent('myCustomEvent', { detail: undefined });

// Will output "null"
console.log(customEvent.detail);
```

Reply:
Thanks for the explanation.

I think we do not need to change the condition, but I would change if you want to.

> "null == undefined" true
> "null != null" false
> "null != undefined" false

Reply:
I got confused by the double (!=) not triple condition (!==). I have changed the my suggestion above. I think using triple condition is less error prone :sweat_smile:

Reply:
Yeah I was thinking about that.

Closed by accident?